internal/host/host.go

```diff
@@ -234,14 +234,22 @@ func (m *Manager) GetNextSteps(ctx context.Context, host *models.Host) (models.S
 }
 
 func (m *Manager) UpdateInstallProgress(ctx context.Context, h *models.Host, progress *models.HostProgress) error {
+	previousProgress := h.Progress
+
+	if previousProgress != nil &&
+		previousProgress.CurrentStage == progress.CurrentStage &&
+		previousProgress.ProgressInfo == progress.ProgressInfo {
+		return nil
+	}
+
 	validStatuses := []string{
 		models.HostStatusInstalling, models.HostStatusInstallingInProgress, models.HostStatusInstallingPendingUserAction,
 	}
 	if !funk.ContainsString(validStatuses, swag.StringValue(h.Status)) {
 		return fmt.Errorf("Can't set progress <%s> to host in status <%s>", progress.CurrentStage, swag.StringValue(h.Status))
 	}
-	previousProgress := h.Progress
-	if h.Progress.CurrentStage != "" && progress.CurrentStage != models.HostStageFailed {
+
+	if previousProgress.CurrentStage != "" && progress.CurrentStage != models.HostStageFailed {
```

Comment:
Isn't this if problematic? don't we set it to empty string on reset?
`if previousProgress.CurrentStage != ""`

Reply:
I'm not sure about what is being set on `reset`.
But it won't interrupt the regular logic flow. This condition is used as an extra verification for using higher stages